connectors/sources/graphql.py

```diff
@@ -380,9 +380,15 @@ def get_default_configuration(cls):
                 "tooltip": "Specifies which GraphQL objects should be indexed as individual documents. This allows finer control over indexing, ensuring only relevant data sections from the GraphQL response are stored as separate documents. Use '.' to provide full path of the object from the root of the response. For example 'organization.users.nodes'",
                 "type": "list",
             },
+            "graphql_field_id": {
+                "label": "GraphQL field name for ID",
+                "order": 10,
+                "tooltip": "Specifies a string field within the document, with the requirement that each document must have a distinct value for this field.",
```

Comment:
Generally looks good, but if we allow to define multiple graphQL objects to index, we should also allow to specify ID for each of those objects, so maybe some approach with comma-separated list of field names?

Reply:
But mostly single source have same name of field to identify the document. Suppose GitHub have unique field `id` so we can found `id` in every object.

Reply:
> But mostly single source have same name of field to identify the document.

Can we guarantee that it holds for all sources? It really depends on the implementation and the user schema what field denotes the unique ID for the object

> Suppose GitHub have unique field id so we can found id in every object.

I think we should be careful overfitting connector to public github graphql server. We need to test the connector with other implementations?

Reply:
Okay in that case if we want to move further with multiple ID we need to take json object as an input to map IDs with GraphQL objects. So IMO we can add IDs json in the same configuration field (`graphql_object_list`).
```
{
      "repository.issue": "id",
      "repository.pullRequest": "id"
}
```
So, WDYT? do you want me to move further with json?

Reply:
Hmm I'm thinking about the aspect of "user friendliness", typing a JSON by hand can be cumbersome, what about two comma-separated lists?

Reply:
But in that case, how can we map IDs with objects? I suppose we need to match the indices of two different lists, and users would need to manage two different config fields, which I find more complex. Additionally, I'm not sure if the framework returns a list in the same order as the input or not. We have many JSON input fields such as variables and headers. In my opinion, most of the users who work with GraphQL can easily configure this.

Reply:
Can we wait with that decision for @danajuratoni? It seems to be a UX decision and should be quick and easy to implement

Reply:
I expect GraphQL users to be comfortable with JSON and I think JSON would be more intuitive than explaining the 2 lists are related, ordered and we'd need to validate the number of values match.
Let's go with JSON!